### ISSUE TITLE
[bug] build fail for SIMULATOR64 with xcode 14

### ISSUE THREAD

**Issue by littleGnAl** [leetal/ios-cmake#164]
**Describe the bug**
build fail for SIMULATOR64 with xcode 14, which works fine with xcode 13.x

**To Reproduce**

1. Clone the https://github.com/AgoraIO-Extensions/iris_event_flutter
2. Run `cxx/build-ios.sh`

```
git clone https://github.com/AgoraIO-Extensions/iris_event_flutter.git
bash cxx/build-ios.sh $(pwd)/cxx $(pwd)/ios
```

**Expected behavior**
The SIMULATOR64 should be built with xcode 14.

**Environment**
* What version of CMake did you run?
```
-- Using Xcode version: 14.0.1
-- CMake version: 3.21.3
-- Using a data_ptr size of: 8
-- Bitcode: Enabled
-- ARC: Enabled
-- Hiding symbols: Enabled```
* Your faulty CMakeLists.txt here:
```
cmake
--- Paste here ---
The `CMakeLists.txt` file: https://github.com/AgoraIO-Extensions/iris_event_flutter/blob/main/cxx/CMakeLists.txt
```

**Comment by leetal** [comment]
Hmm.. We're building with Xcode 14 without any problems whatsoever. I'll look into this the coming day(s)!

**Comment by littleGnAl** [comment]
Thanks for the reply, this issue be fixed after upgrading the cmake to `3.24.3`.
